test/integration/hostgroup_test.rb

```diff
@@ -22,7 +22,7 @@ class HostgroupIntegrationTest < ActionDispatch::IntegrationTest
   end
 
   test 'edit shows errors on invalid lookup values' do
-    group = FactoryGirl.create(:hostgroup, :with_puppetclass)
+    group = FactoryGirl.create(:hostgroup, :with_puppetclass, :with_environment)
```

Comment:
Keeping with_puppetclass with an environment too will avoid having to change every test, implicit is fine for this!